jenkins_test.go

```diff
@@ -6,21 +6,35 @@ import (
 	"os"
 	"testing"
 	"time"
-
 	"github.com/stretchr/testify/assert"
 )
 
 var (
 	jenkins *Jenkins
 )
 
+const (
+	BaseURL = "http://localhost:8080"
+	UserName = "admin"
+	Password = "admin"
+)
+
+func createClient(t *testing.T) {
+	if jenkins == nil {
+		jenkins = New(BaseURL, UserName, Password)
+		_, err := jenkins.Init()
+		assert.Nil(t, err)
+	}
+}
+
 func TestInit(t *testing.T) {
-	jenkins = CreateJenkins("http://localhost:8080", "admin", "admin")
+	createClient(t)
```

Comment:
I was working on something more like an idempotent setup.  It feels like there's a shared resource/global there, so I started with a goal to use that to set up something more general, and not re-use or potentially cause issues with shared memory.  

Additionally, I'm not sure why, but I know that some of these tests fail on the same Jenkins instance. While they pass on a Docker container without any state, they don't pass in a shared environment, so I was trying to figure out what might not be tested as far as implicit assumptions about the previous state, so I started down that path.  I haven't completed it, though, so it's technically a WIP.